rest/blip_client_test.go

```diff
@@ -1135,3 +1143,60 @@ func (btc *BlipTesterCollectionClient) sendPushMsg(msg *blip.Message) error {
 	btc.addCollectionProperty(msg)
 	return btc.parent.pushReplication.sendMsg(msg)
 }
+
+func (btc *BlipTesterClient) BlipClientCollectionSetup(collection *db.DatabaseCollection) (*BlipTesterCollectionClient, error) {
+	var btcCollection *BlipTesterCollectionClient
+	scopeAndCollectionKey := strings.Join([]string{collection.ScopeName(), collection.Name()}, base.ScopeCollectionSeparator)
+	if !base.IsDefaultCollection(collection.ScopeName(), collection.Name()) {
+		checkpointID1 := "checkpoint1"
+		getCollectionsRequest, err := db.NewGetCollectionsMessage(db.GetCollectionsRequestBody{
+			CheckpointIDs: []string{checkpointID1},
+			Collections:   []string{fmt.Sprintf("%s.%s", collection.ScopeName(), collection.Name())},
+		})
+		if err != nil {
+			return nil, err
+		}
+		err = btc.pushReplication.sendMsg(getCollectionsRequest)
+		if err != nil {
+			return nil, err
+		}
+		resp := getCollectionsRequest.Response()
+		if resp == nil {
+			return nil, fmt.Errorf("error in getCollections response")
+		}
```

Comment:
We should read the response body to make sure the other side of the replication agreed to replicate with the given collection.

There's detail about the response here or here:
- https://github.com/couchbase/couchbase-lite-core/blob/master/modules/docs/pages/replication-protocol.adoc#getCollections
- https://docs.google.com/document/d/1kbPUuWgtnAgV9Oj55_soWmIPA8f9v3RBrjMXVvbn4p4/edit#heading=h.v2cz3nwptjc4